src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -23,101 +23,196 @@ public DataProcessor(List<RoadAccident> roadAccidentList){
+            result.put(roadSurfaceConditions, newValue);
+        }
+
+        return result;
     }
 
     /**
      * find the weather conditions which caused the top 3 number of incidents
      * as example if there were 10 accidence in rain, 5 in snow, 6 in sunny and 1 in foggy, then your result list should contain {rain, sunny, snow} - top three in decreasing order
+     *
      * @return
      */
-    public List<String> getTopThreeWeatherCondition7(){
-        return null;
+    public List<String> getTopThreeWeatherCondition7() {
+        List<String> result = new ArrayList<>();
+
+        Map<String, Long> accidentsByWeatherConditions = new HashMap<>();
+
+        Iterator<RoadAccident> roadAccidentIterator = roadAccidentList.iterator();
+
+        while (roadAccidentIterator.hasNext()) {
+            RoadAccident current = roadAccidentIterator.next();
+            String weatherConditions = current.getWeatherConditions();
+
+            Long newValue = accidentsByWeatherConditions.get(weatherConditions) == null ? 1 : Long.valueOf(accidentsByWeatherConditions.get(weatherConditions) + 1);
+            accidentsByWeatherConditions.put(weatherConditions, newValue);
+        }
+
+        List<Map.Entry<String, Long>> listAccidentsByConditions = new LinkedList<Map.Entry<String, Long>>(accidentsByWeatherConditions.entrySet());
+
+        Collections.sort(listAccidentsByConditions, new Comparator<Map.Entry<String, Long>>(){
+
+            @Override
+            public int compare(Map.Entry<String, Long> o1, Map.Entry<String, Long> o2) {
+                // Decreasing order
+                return o2.getValue().compareTo(o1.getValue());
+            }
+        });
+
+        List<Map.Entry<String, Long>> topAccidents = listAccidentsByConditions.subList(0, 3);
+
+        for (Map.Entry<String, Long>  accident: topAccidents) {
+            result.add(accident.getKey());
+        }
+
+        return result;
     }
 
     /**
      * return a multimap where key is a district authority and values are accident ids
      * ex:
      * authority1 -> id1, id2, id3
      * authority2 -> id4, id5
+     *
      * @return
      */
-    public Multimap<String, String> getAccidentIdsGroupedByAuthority7(){
-        return null;
+    public Multimap<String, String> getAccidentIdsGroupedByAuthority7() {
+        Multimap<String, String> result = ArrayListMultimap.create();
+        Iterator<RoadAccident> roadAccidentIterator = roadAccidentList.iterator();
+
+        while (roadAccidentIterator.hasNext()) {
+            RoadAccident current = roadAccidentIterator.next();
+            result.put(current.getDistrictAuthority(), current.getAccidentId());
+        }
+
+        return result;
     }
 
 
     // Now let's do same tasks but now with streaming api
 
 
+    public RoadAccident getAccidentByIndex(String index) {
+        Optional<RoadAccident> result = roadAccidentList.stream()
+                .filter(acc -> acc.getAccidentId().equals(index))
```

Comment:
More about clean code practice - what is acc? With modern IDE it's not taking long time to use name like roadAccident. Easy to read - easy to refactor ;)